### PR TITLE
WIP: track events  using matomo

### PR DESCRIPTION
**Pull request by mrsimpson**
No idea why, but it's not being debounced

### CODE REVIEW

File: components.d.ts [hunk 1]
```diff
@@ -16,9 +16,6 @@ declare module '@vue/runtime-core' {
     NForm: typeof import('naive-ui')['NForm']
     NFormItem: typeof import('naive-ui')['NFormItem']
     NIcon: typeof import('naive-ui')['NIcon']
-    NInput: typeof import('naive-ui')['NInput']
-    NInputGroup: typeof import('naive-ui')['NInputGroup']
-    NInputGroupLabel: typeof import('naive-ui')['NInputGroupLabel']
```
Comment (jschirrmacher) [inline components.d.ts:21]: Why did you modify the d.ts files for naive-ui? Has is anything to do with tracking?

File: src/components/TheConsumptionParameters.vue [hunk 2]
```diff
@@ -68,7 +70,15 @@ function setSavings(savingsPercent: number) {
             :min="1"
             :step="1000"
             :validator="validatePositive"
-            :on-input="emit('consumptionChanged', model.consumption)"
```
Comment (jschirrmacher) [inline src/components/TheConsumptionParameters.vue:71]: A single call to `emit()` while rendering the template at least could not have worked.

File: src/components/TheConsumptionParameters.vue [hunk 3]
```diff
@@ -68,7 +70,15 @@ function setSavings(savingsPercent: number) {
...
+            :on-input="
+              debounce(
+                () => {
+                  trackInput('consumption') &&
+                  emit('consumptionChanged', model.consumption)
+                },
+                2000
+              )
+            "
```
Comment (jschirrmacher) [inline src/components/TheConsumptionParameters.vue]: I understand, that the`debounce()` function is a HOF, thus returning a function, so that might work at least.

However, according to https://www.naiveui.com/en-US/os-theme/components/input-number there is no such property `on-input`.

Maybe `on-update:value` would work?

**Review by jschirrmacher — DISMISSED** [review]

File: components.d.ts (hunk 1 above)
Comment (mrsimpson) [reply components.d.ts:21]: I didn't modify it: this is generated by the [auto-import](https://github.com/antfu/unplugin-auto-import).

Comment (mrsimpson) [reply components.d.ts:21]: It's got nothing to do with the current issue

File: src/components/TheConsumptionParameters.vue (hunk 2 above)
Comment (mrsimpson) [reply src/components/TheConsumptionParameters.vue:71]: see next comment below

File: src/components/TheConsumptionParameters.vue (hunk 3 above)
Comment (mrsimpson) [reply src/components/TheConsumptionParameters.vue]: Ok, this has been som trial-and-lots-of-errors.
tl;dr: I always wanted to use `on-update:value` but this lead to the input not reacting on interactions with the +/-
[`on-input`](https://www.naiveui.com/en-US/os-theme/components/input#Input-Props) was triggered however.

Bottom line: using `ref` and reactive state passed as props requires props to be watched explicitly (see [this comment](https://github.com/mrsimpson/energy/pull/89/commits/da3fc51397144f3805959e32e0c7ec7765e88034#diff-a104f881632b86b395547ee5d44e6e4fb8dba269f55e8eae4c11360c6caf1399R25-R27))
At least now, it looks fine to me

**Comment by mrsimpson** [comment]
@jschirrmacher I changed the model handling so that it's possible not to always use `@update:value`.
I also tracked all inputs (without passing sensitive consumption data) and debounced this interaction.
As a result of this
- The `+/-` buttons are also debounced
- The input only has an effect after the debounce (which I consider ok)
- When using `@input:value`, a `blur` event is needed prior to the input having any effect. I now changed this `@changed` so that even after typing and when remaining in the same input, values are being updated
